tests/test_attributes.py

```diff
@@ -0,0 +1,96 @@
+import httpx
+import pytest
+from fastapi import status
+
+from app.api import utility as util
+
+
+def test_partially_failed_terms_fetching_handled_gracefully(
+    test_app, monkeypatch
+):
+    """
+    Test that when getting term instances for an attribute (/attribute/{data_element_URI}) fails for some nodes due to an error,
+    the overall API get request still succeeds, and the response includes a list of the encountered errors along with the successfully fetched terms.
+    """
+    monkeypatch.setattr(
+        util,
+        "FEDERATION_NODES",
+        {
+            "https://firstpublicnode.org/": "First Public Node",
+            "https://secondpublicnode.org/": "Second Public Node",
+        },
+    )
```

Comment:
I guess we'll need this for all the tests, right? Maybe we should lift all this into a fixture itself, see last example in https://docs.pytest.org/en/6.2.x/monkeypatch.html#monkeypatching-returned-objects-building-mock-classes